include/h2o.h

```diff
@@ -503,6 +558,11 @@ struct st_h2o_context_t {
         h2o_timestamp_string_t *value;
     } _timestamp_cache;
 
+    /**
+     * counter for errors internally emitted by h2o
+     */
+    h2o_emitted_errors_t emitted_errors;
```

Comment:
I believe the second of the two new commits address what you suggested. I had to keep a few calls directly to `h2o_send_error_generic`, since sometimes the error isn't directly known in the call site.
Thanks!